### DISCUSSION TITLE
How to add negspacy to spacy config.cfg file?

### DISCUSSION THREAD

**Discussion by canberkarc** [jenojp/negspacy#48] [Q&A]
I'm trying to add negex to the pipeline in config.cfg file which is taken from https://spacy.io/usage/training but I couldn't find a way to do that. Can you please help me? Thanks in advance.

**Comment by jenojp** [comment]
Hi there - could you share the .CFG file or excerpt that you tried so far? 

I believe you add the pipeline by name to the `pipeline` list as seen in the example here: https://spacy.io/api/data-formats#config-nlp
```
[nlp]
pipeline = ["sentencizer", "ner", "entity_linker"]

[components.ner]
source = "en_core_web_sm"

[training]
frozen_components = ["ner"]
annotating_components = ["sentencizer", "ner"]
```


```4. Include a non-trainable component, like a rule-based [EntityRuler](https://spacy.io/api/entityruler) or [Sentencizer](https://spacy.io/api/sentencizer), or a fully [custom component](https://spacy.io/usage/processing-pipelines#custom-components).```

would be analogous I believe.

**Comment by canberkarc** [comment]
Hello, since I couldn't add negex to pipeline in .cfg file, I did it manually by using non-trainable components.

My pipeline is as given below:

```

nlp = spacy.blank("tr")

lemmatizer = nlp.add_pipe("lemmatizer")
lemmatizer.initialize()
nlp.add_pipe("sentencizer")
ruler = nlp.add_pipe("entity_ruler")

ruler.add_patterns(patterns)
nlp.add_pipe("negex", config={"ent_types":["PERSON","ORG"]})


```
